SignalServiceKit/tests/Contacts/SignalRecipientTest.m

```diff
@@ -26,32 +27,39 @@ @implementation SignalRecipientTest
 - (void)setUp
 {
     [super setUp];
+
     self.localNumber = @"+13231231234";
     [[TSAccountManager sharedInstance] storeLocalNumber:self.localNumber];
 }
 
+- (void)tearDown
+{
+    [super tearDown];
+}
+
 - (void)testSelfRecipientWithExistingRecord
 {
     // Sanity Check
     XCTAssertNotNil(self.localNumber);
-    [[[SignalRecipient alloc] initWithTextSecureIdentifier:self.localNumber relay:nil] save];
-    XCTAssertNotNil([SignalRecipient recipientWithTextSecureIdentifier:self.localNumber]);
 
-    SignalRecipient *me = [SignalRecipient selfRecipient];
-    XCTAssert(me);
-    XCTAssertEqualObjects(self.localNumber, me.uniqueId);
+    [OWSPrimaryStorage.sharedManager.dbReadWriteConnection
+        readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+            [SignalRecipient markRecipientAsRegisteredAndGet:self.localNumber transaction:transaction];
+
+            XCTAssertTrue([SignalRecipient isRegisteredRecipient:self.localNumber transaction:transaction]);
+        }];
 }
 
 - (void)testSelfRecipientWithoutExistingRecord
```

Comment:
IMO this test can probably go away since you got rid of the `selfRecipient` method.

Reply:
Removed.